packages/@react-spectrum/listbox/docs/ListBox.mdx

```diff
@@ -28,35 +35,313 @@ category: Collections
+  let [frequency, setFrequency] = React.useState();
+
+  return (
+    <>
+      <ListBox
+        aria-label="Choose frequency"
+        selectionMode="single"
+        onSelectionChange={selected => setFrequency(selected)}
+        width="size-2400">
+        <Item key="Rarely">Rarely</Item>
+        <Item key="Sometimes">Sometimes</Item>
+        <Item key="Always">Always</Item>
+      </ListBox>
+      <p>You selected: {frequency}</p>
+    </>
+  );
+}
+```
+
+When using ListBox with dynamic items, selection works much the same way using `key`. However, if your data already has an `id` property (as is common with many data sets), ListBox can use that id without needing to specify a `key` prop. The below example shows ListBox using the id of each item from the `items` array as the selected value without the need for `key`. Note that `key` will always take precedence if set.
+
+```tsx example
+function DynamicExample() {
+  let [animalId, setAnimalId] = React.useState();
+  let options = [
+    {id: 1, name: 'Aardvark'},
+    {id: 2, name: 'Cat'},
+    {id: 3, name: 'Dog'},
+    {id: 4, name: 'Kangaroo'},
+    {id: 5, name: 'Koala'},
+    {id: 6, name: 'Penguin'},
+    {id: 7, name: 'Snake'},
+    {id: 8, name: 'Turtle'},
+    {id: 9, name: 'Wombat'}
+  ];
+
+  return (
+    <>
+      <ListBox
+        selectionMode="single"
+        aria-label="Pick an animal"
+        items={options}
+        onSelectionChange={selected => setAnimalId(selected)}
+        width="size-2400">
+        {item => <Item>{item.name}</Item>}
+      </ListBox>
+      <p>Your favorite animal has id: {animalId}</p>
+    </>
+  );
+}
+```
+
+## Complex Items
+Items within ListBox also allow for additional content used to better communicate options. Icons and descriptions can be added to the `children` of `Item` as shown in the example below. If a description is added, the prop `slot="description"` must be used to distinguish the different `<Text>` elements.
+
+```tsx example
+<ListBox width="size-2400" aria-label="Options" selectionMode="single">
+  <Section title="Permission">
+    <Item textValue="Read">
+      <Book size="S" />
+      <Text>Read</Text>
+      <Text slot="description">Read Only</Text>
+    </Item>
+    <Item textValue="Write">
+      <Draw size="S" />
+      <Text>Write</Text>
+      <Text slot="description">Read and Write Only</Text>
+    </Item>
+    <Item textValue="Admin">
+      <BulkEditUsers size="S" />
+      <Text>Admin</Text>
+      <Text slot="description">Full access</Text>
+    </Item>
+  </Section>
+</ListBox>
+```
+## Asynchronous loading
+
+ListBox supports loading data asynchronously, and will display a progress circle when the `isLoading` prop is set.
+It also supports infinite scrolling to load more data on demand as the user scrolls, via the `onLoadMore` prop.
+
+This example uses the [useAsyncList](../react-stately/useAsyncList.html) hook to handle loading the data.
+See the docs for more information.
+
+```tsx example
```

Comment:
not sure what's happening here, but for me the entire list of pokemon is being loaded immediately